frontends/concrete-python/concrete/fhe/compilation/module.py

```diff
@@ -0,0 +1,669 @@
+
+    @property
+    def encrypted_negation_count(self) -> int:
+        """
+        Get the number of encrypted negations in the function.
+        """
+        return self.runtime.server.encrypted_negation_count(self.name)  # pragma: no cover
+
+    @property
+    def encrypted_negation_count_per_parameter(self) -> Dict[Parameter, int]:
+        """
+        Get the number of encrypted negations per parameter in the function.
+        """
+        return self.runtime.server.encrypted_negation_count_per_parameter(
+            self.name
+        )  # pragma: no cover
+
+    @property
+    def encrypted_negation_count_per_tag(self) -> Dict[str, int]:
+        """
+        Get the number of encrypted negations per tag in the function.
+        """
+        return self.runtime.server.encrypted_negation_count_per_tag(self.name)  # pragma: no cover
+
+    @property
+    def encrypted_negation_count_per_tag_per_parameter(self) -> Dict[str, Dict[Parameter, int]]:
+        """
+        Get the number of encrypted negations per tag per parameter in the function.
+        """
+        return self.runtime.server.encrypted_negation_count_per_tag_per_parameter(
+            self.name
+        )  # pragma: no cover
+
+    @property
+    def statistics(self) -> Dict:
+        """
+        Get all statistics of the function.
+        """
+        attributes = [
+            "size_of_inputs",
+            "size_of_outputs",
+            "programmable_bootstrap_count",
+            "programmable_bootstrap_count_per_parameter",
+            "programmable_bootstrap_count_per_tag",
+            "programmable_bootstrap_count_per_tag_per_parameter",
+            "key_switch_count",
+            "key_switch_count_per_parameter",
+            "key_switch_count_per_tag",
+            "key_switch_count_per_tag_per_parameter",
+            "packing_key_switch_count",
+            "packing_key_switch_count_per_parameter",
+            "packing_key_switch_count_per_tag",
+            "packing_key_switch_count_per_tag_per_parameter",
+            "clear_addition_count",
+            "clear_addition_count_per_parameter",
+            "clear_addition_count_per_tag",
+            "clear_addition_count_per_tag_per_parameter",
+            "encrypted_addition_count",
+            "encrypted_addition_count_per_parameter",
+            "encrypted_addition_count_per_tag",
+            "encrypted_addition_count_per_tag_per_parameter",
+            "clear_multiplication_count",
+            "clear_multiplication_count_per_parameter",
+            "clear_multiplication_count_per_tag",
+            "clear_multiplication_count_per_tag_per_parameter",
+            "encrypted_negation_count",
+            "encrypted_negation_count_per_parameter",
+            "encrypted_negation_count_per_tag",
+            "encrypted_negation_count_per_tag_per_parameter",
+        ]
+        return {attribute: getattr(self, attribute) for attribute in attributes}
+
+
+class FheModule:
```

Comment:
How are modules deployed in a client/server architecture :thinking: It'll be the first thing the ML team will want :stuck_out_tongue:

Reply:
If it's possible already, a test would be nice to make sure it's working.

Reply:
And documentation ofc

Reply:
Indeed, I did not implement all the serialization and deployment related functions, you are totally right. Is that OK for you if I implement it in a upcoming PR  ?

Reply:
Yes, as you wish!